src/cockpit/channels/packages.py

```diff
@@ -108,14 +94,10 @@ def do_done(self) -> None:
 
         else:
             self.send_message(status=200, reason='OK', headers=out_headers)
-            threading.Thread(args=(asyncio.get_running_loop(), document.data),
-                             target=self.send_document_data,
-                             daemon=True).start()
-
-    def send_document_data(self, loop, data):
-        # split data into 4K blocks, to not overwhelm the channel
-        block_size = 4096
-        for i in range(0, len(data), block_size):
-            loop.call_soon_threadsafe(self.send_data, data[i:i + block_size])
-        loop.call_soon_threadsafe(self.done)
-        loop.call_soon_threadsafe(self.close)
+
+            # split data into 4K blocks, to not overwhelm the channel
+            block_size = 4096
```

Comment:
In any case, this is the wrong choice — the flow control only activates at 16k boundaries.  You want `Channel.BLOCK_SIZE` here.